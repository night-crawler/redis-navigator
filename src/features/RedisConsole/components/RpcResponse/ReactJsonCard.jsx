import React from 'react';
import ReactJson from 'react-json-view';
import { Card } from 'semantic-ui-react';
import PropTypes from 'prop-types';


ReactJsonCard.propTypes = {
    result: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    groupArraysAfterLength: PropTypes.number,
};
export default function ReactJsonCard(props) {
    const { result, groupArraysAfterLength = 20 } = props;
    return (
        <Card fluid={ true }>
            <Card.Content header='Object' />
            <Card.Description>
                <ReactJson
                    src={ result }
                    groupArraysAfterLength={ groupArraysAfterLength }
                    name={ false }
                />
            </Card.Description>
        </Card>
    );
}
