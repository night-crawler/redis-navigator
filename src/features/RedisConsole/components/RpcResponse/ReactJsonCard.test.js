/* eslint-disable react/react-in-jsx-scope,no-undef */
import ReactJsonCard from './ReactJsonCard';


describe('ReactJsonCard', () => {
    it('should render ReactJsonCard', () => {
        const result = { a: 1 };
        const wrapper = shallow(<ReactJsonCard result={ result } />);
        expect(wrapper).toMatchSnapshot();
    });
});
