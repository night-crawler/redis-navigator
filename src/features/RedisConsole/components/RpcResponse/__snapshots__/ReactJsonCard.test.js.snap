// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactJsonCard should render ReactJsonCard 1`] = `
<Card
  fluid={true}
>
  <CardContent
    header="Object"
  />
  <CardDescription>
    <t
      groupArraysAfterLength={20}
      name={false}
      src={
        Object {
          "a": 1,
        }
      }
    />
  </CardDescription>
</Card>
`;
